### ISSUE TITLE
Web based menu

### ISSUE THREAD

**Issue by mstormi** [openhab/openhabian#1180]
Following up on https://github.com/orgs/openhab/teams/openhabian-maintainers/discussions/3, we should "upgrade" openhabian-config to provide a Web interface rather than the current shell CLI (or in addition to it).
This will allow users to get along without SSH, bash etc.
It'll also allow for greater UX since we can integrate with OH's admin GUI.

As an intermediate step to take effect as per today already, all features should be made to work in 'unattended' AND 'interactive' mode. They **must not** retrieve individual values from openhabian.conf but leave that to a routine at the very beginning to source the complete conf file and set all variables in one go. This part then is to be replaced by the web interface.
Potentially any web framework would then "just" need to collect input from users (more or less copy'n paste what the current code does in interactive mode) to set the variables and finally execute the existing install routines in unattended mode.

For your attention: I've updated [CONTRIBUTING.md](https://github.com/openhab/openhabian/blob/master/CONTRIBUTING.md).

Any helping hand from a web frontend designer will happily be taken !

**Comment by paulianttila** [comment]
Maybe e.g. https://cockpit-project.org could be used as a framework (Red Hat sponsored project). 

It give basic information out of the box and it can be extended.

- https://cockpit-project.org/blog/cockpit-starter-kit.html
- https://cockpit-project.org/blog/creating-plugins-for-the-cockpit-user-interface.html

>t'll also allow for greater UX since we can integrate with OH's admin GUI.

Not sure about this, but it should be possible as well

- https://cockpit-project.org/guide/latest/embedding.html

**Comment by HazeHunter11** [comment]
Regarding an admin GUI. I‘m not sure if it makes sense to build a admin GUI from stretch. Do you know webmin, it is a nice tool that is widely used? This has already great system features like an web based shell or performance measurement data on a dashboard.

I would suggest to use webmin and customize this. Remove unneeded modules and develop own custom modules that offer the features from the ch-cli. Maybe also add some log viewer in it.

**Comment by WolfgangSn** [comment]
Have you thought about using shell-in-box ( https://code.google.com/archive/p/shellinabox/ ) ?
see also https://www.tecmint.com/shell-in-a-box-a-web-based-ssh-terminal-to-access-remote-linux-servers/
Assumed that this works you could maintain one script and use it via ssh access as well as in the browser.

**Comment by mstormi** [comment]
FYI work ongoing https://community.openhab.org/t/web-ui-for-openhabian/113418/12

**Comment by mstormi** [comment]
I assume no feedback in various months means no interest in getting this finalized?

Any input, anyone ?

**Comment by tflyx** [comment]
The feature would bei great to have. Unfortunately my linux skills are to bad to support build IT.

**Comment by miloit** [comment]
Is this followed up somehow?

**Comment by ecdye** [comment]
No, you are welcome to work on this if you would like. I don't have the time nor expertise to create something like this right now. While it is an excellent suggestion, it is not a super high priority at the moment.

**Comment by mstormi** [comment]
I don't think is is gonna happen.
Feel free to reopen if there's news.
